fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/CompletionExceptionLogHelper.java

```diff
@@ -78,9 +99,55 @@ private static Throwable asCauseThrowable(@Nonnull Throwable ex) {
         if (cause == null) {
             return ex;
         }
+
         if (addSuppressed) {
-            cause.addSuppressed(ex);    // Remember the original.
+            synchronized (cause) {
+                Throwable[] suppressedExceptions = cause.getSuppressed();
+                if (suppressedExceptions.length < maxSuppressedCount) {
+                    cause.addSuppressed(ex);
+                } else {
+                    IgnoredSuppressedExceptionCount suppressedCount;
+                    int lastIndex = suppressedExceptions.length - 1;
```

Comment:
Oh, I think this has a problem if `maxSuppressedCount` is set to zero or something negative. We should either (1) see if `suppressedExceptions` is empty or (2) validate that `maxSuppressedCount` is positive.

Reply:
Right, fixed.